### PR TITLE
🚧Refactor unit test assertions

### LINKED ISSUE #106: 🚧Refactor unit test assertions
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Refactor all of the unit test assertions to use the **FluentAssertions** library.
This will greatly cleanup the unit testing side of things and make the creation of unit tests easier.

> **Note**
> 1. We have been migrating all of our assertion code for the past 6 months from standard XUnit assertions API to [FluentAssertions API](https://fluentassertions.com/introduction) API.  Most of the code base has been converted and examples can be found all of the place.  Please don't hesitate to ask questions and for help if needed.
> 2. Please follow our [Contribution Guide](https://github.com/KinsonDigital/.github/blob/main/docs/CONTRIBUTING.md)
> 3. Join the [KinsonDigital Community](https://discord.gg/qewu6fNgv7)!!

### Acceptance Criteria

- [x] Latest version of FluentAssertions Nuget package added to the project
- [x] Refactor the files below: 
  - [x] `AudioDeviceManagerNotInitializedExceptionTests.cs` - 000 lines
  - [x] `InitializeDeviceExceptionTests.cs` - 50 lines
  - [x] `AudioDeviceManagerTests.cs` - 60 lines
  - [x] `AudioExceptionTests.cs` - 509 lines
  - [x] `LoadLibraryExceptionTests.cs` - 50 lines

### ToDo Items

- [x] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [x] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [x] Issue linked to the correct project _(if required)_.
- [x] Issue linked to the correct milestone _(if required)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by thestbar**
### **Description:**

To allow this pull request to be merged, please make sure that the following items of this pull request are complete.

This pull request closes #106

✅The pull request head branch must be a feature branch with the syntax 'feature/<issue-num>-sync-testing'.  
✅The pull request base branch must be a 'main' or 'preview' branch..  
✅The pull request head branch contains a valid issue number.  
✅The pull request title matches the linked issue title exactly.  
✅The pull request assignees match the assignees of the issue.  
✅The pull request labels match the labels of the issue.  
✅The pull request organizational projects match the organizational projects of the issue.  
✅The pull requests milestone matches the milestone of the issue.  

---

### **Additional Info:**
1. Pull requests are automatically synced with the associated issue upon creation by one of the KinsonDigital workflow bots. 
2. The associated issue is the issue number that is embedded in the pull request head branch.
3. The list above will be automatically updated as the pull request's various settings match or do not match the associated issue.
4. To manually sync the pull request to the issue, create a comment with the `[run-sync]` command.
	> **Note** You must be an admin member of the organization to use this command.

---

- [x] Sync with the issue.  Use this to enable or disable this pull request from syncing with its associated issue.

### CODE REVIEW

**Comment by thestbar** [comment]
After discussion with @CalvinWilkinson the PR will be pushed for review even if sync status check fails, because the reason of the failure has to do with the CICD system.

**Comment by CalvinWilkinson** [comment]
[run-sync]

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs [hunk 1]
```diff
@@ -21,31 +22,35 @@ public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
...
     {
         // Act
-        var exception = new AudioDeviceManagerNotInitializedException("test-message");
+        var exceptionMessage = "test-message";
+        var exception = new AudioDeviceManagerNotInitializedException(exceptionMessage);
 
         // Assert
-        Assert.Equal("test-message", exception.Message);
+        exception.Message.Should().Be(exceptionMessage);
     }
```
Comment (CalvinWilkinson) [inline Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Please refactor to be more aligned with the AAA (triple A) unit test layout.

**Reason:** This is to help improve the consistency and readability of the tests.

Suggestion:
```cs
    [Fact]
    public void Ctor_WhenInvokedWithSingleMessageParam_CorrectlySetsMesage()
    {
        // Arrange
        var expected = "test-message";

        // Act
        var exception = new AudioDeviceManagerNotInitializedException(expected);

        // Assert
        exception.Message.Should().Be(expected);
    }
```

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs [hunk 2]
```diff
@@ -21,31 +22,35 @@ public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
...
-        var deviceException = new AudioDeviceManagerNotInitializedException("test-exception", innerException);
+        var exceptionMessage = "test-exception";
+        var deviceException = new AudioDeviceManagerNotInitializedException(exceptionMessage, innerException);
 
         // Assert
-        Assert.Equal("inner-exception", deviceException.InnerException.Message);
-        Assert.Equal("test-exception", deviceException.Message);
+        deviceException.InnerException.Message.Should().Be(innerExceptionMessage);
+        deviceException.Message.Should().Be(exceptionMessage);
     }
```
Comment (CalvinWilkinson) [inline Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Please refactor to be more aligned with the AAA (triple A) unit test layout.

**Reason:** This is to help improve the consistency and readability of the tests.

**Suggestion:**
```cs
    [Fact]
    public void Ctor_WhenInvokedWithMessageAndInnerException_ThrowsException()
    {
        // Arrange
        var expectedInnerExceptionMessage = "inner-exception";
        var expectedExceptionMessage = "test-exception";
        var innerException = new Exception(expectedInnerExceptionMessage);

        // Act
        var deviceException = new AudioDeviceManagerNotInitializedException(expectedExceptionMessage, innerException);

        // Assert
        deviceException.InnerException.Message.Should().Be(expectedInnerExceptionMessage);
        deviceException.Message.Should().Be(expectedExceptionMessage);
    }
```

File: Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs [hunk 3]
```diff
@@ -21,31 +22,35 @@ public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
         var exception = new InitializeDeviceException();
 
         // Assert
-        Assert.Equal("There was an issue initializing the audio device.", exception.Message);
+        var expectedMessage = "There was an issue initializing the audio device.";
+        exception.Message.Should().Be(expectedMessage);
```
Comment (CalvinWilkinson) [inline Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs]: Please refactor this test to follow AAA (triple A) unit test layout

**Reason:** This is to improve consistency and readability of the unit tests across the code base.

```cs
    [Fact]
    public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
    {
        // Arrange
        var expected = "There was an issue initializing the audio device.";

        // Act
        var exception = new InitializeDeviceException();

        // Assert
        exception.Message.Should().Be(expected);
    }
```

File: Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs [hunk 4]
```diff
@@ -21,31 +22,35 @@ public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
...
         // Act
-        var deviceException = new InitializeDeviceException("test-exception", innerException);
+        var exceptionMessage = "test-exception";
+        var deviceException = new InitializeDeviceException(exceptionMessage, innerException);
 
         // Assert
-        Assert.Equal("inner-exception", deviceException.InnerException.Message);
-        Assert.Equal("test-exception", deviceException.Message);
+        deviceException.InnerException.Message.Should().Be(innerExceptionMessage);
+        deviceException.Message.Should().Be(exceptionMessage);
```
Comment (CalvinWilkinson) [inline Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs]: Please refactor this test to follow the AAA (triple A) unit test layout

**Reason:** This is to improve the consistency and readability of the unit tests across the code base.

**Suggestion:**
```cs
    [Fact]
    public void Ctor_WhenInvokedWithMessageAndInnerException_ThrowsException()
    {
        // Arrange
        var expectedInnerExceptionMessage = "inner-exception";
        var expectedExceptionMessage = "test-exception";
        var innerException = new Exception(expectedInnerExceptionMessage);

        // Act
        var deviceException = new InitializeDeviceException(expectedExceptionMessage, innerException);

        // Assert
        deviceException.InnerException.Message.Should().Be(expectedInnerExceptionMessage);
        deviceException.Message.Should().Be(expectedExceptionMessage);
    }
```

File: Testing/CASLTests/Exceptions/AudioExceptionTests.cs
Comment (CalvinWilkinson) [inline Testing/CASLTests/Exceptions/AudioExceptionTests.cs:1]: Refactor tests in this file to follow the AAA (triple A) just like it is shown in previous comments.

File: Testing/CASLTests/Exceptions/LoadLibraryExceptionTests.cs
Comment (CalvinWilkinson) [inline Testing/CASLTests/Exceptions/LoadLibraryExceptionTests.cs:1]: Refactor tests in this file to follow the AAA (triple A) just like it is shown in previous comments.

**Review by CalvinWilkinson — CHANGES_REQUESTED** [review]

**Comment by CalvinWilkinson** [comment]
[run-sync]

File: Testing/CASLTests/Exceptions/LoadLibraryExceptionTests.cs
Comment (thestbar) [reply Testing/CASLTests/Exceptions/LoadLibraryExceptionTests.cs:1]: Refactored tests to follow triple A test layout.

File: Testing/CASLTests/Exceptions/AudioExceptionTests.cs
Comment (thestbar) [reply Testing/CASLTests/Exceptions/AudioExceptionTests.cs:1]: Refactored tests to follow triple A test layout.

File: Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs (hunk 4 above)
Comment (thestbar) [reply Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs]: Refactored test to follow triple A test layout.

File: Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs (hunk 3 above)
Comment (thestbar) [reply Testing/CASLTests/Devices/Exceptions/InitializeDeviceExceptionTests.cs]: Refactored test to follow triple A test layout.

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs (hunk 2 above)
Comment (thestbar) [reply Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Refactored test to follow triple A test layout.

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs (hunk 1 above)
Comment (thestbar) [reply Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Refactored test to follow triple A test layout.

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs [hunk 5]
```diff
@@ -21,31 +22,37 @@ public void Ctor_WhenInvokedWithNoParam_CorrectlySetsMessage()
         var exception = new AudioDeviceManagerNotInitializedException();
 
         // Assert
-        Assert.Equal("The audio device manager has not been initialized.", exception.Message);
+        var expectedMessage = "The audio device manager has not been initialized.";
+        exception.Message.Should().Be(expectedMessage);
     }
 
     [Fact]
     public void Ctor_WhenInvokedWithSingleMessageParam_CorrectlySetsMesage()
```
Comment (KinsonDigitalAdmin) [inline Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Whoops!!  I just caught a spelling mistake!!

Could you possibly fix this real quick?

```suggestion
    public void Ctor_WhenInvokedWithSingleMessageParam_CorrectlySetsMessage()
```

I actually did a search in the entire code base and found multiple instances of this spelling mistake.  I had a hunch that it was copied from exception tests to other exception tests and I was correct.  I found 6 other instances of this unit test name with the same spelling mistake.

If you just do a search for `Ctor_WhenInvokedWithSingleMessageParam_CorrectlySetsMesage`, you will find them all.

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

File: Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs (hunk 5 above)
Comment (thestbar) [reply Testing/CASLTests/Devices/Exceptions/AudioDeviceManagerNotInitializedExceptionTests.cs]: Fixed the spelling mistake on all occurrences!

**Review by CalvinWilkinson — APPROVED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**